operator/cmd/run/run.go

```diff
@@ -501,6 +528,47 @@ func Run(
 		}
 	}
 
+	if enableStatefulSetDecommissioner {
+		d := decommissioning.NewStatefulSetDecommissioner(mgr, &v1Fetcher{client: mgr.GetClient()},
+			decommissioning.WithCleanupPVCs(false),
+			decommissioning.WithFactory(internalclient.NewFactory(mgr.GetConfig(), mgr.GetClient())),
+			decommissioning.WithFilter(func(ctx context.Context, sts *appsv1.StatefulSet) (bool, error) {
+				if !slices.ContainsFunc(
+					sts.OwnerReferences,
+					func(ownerRef metav1.OwnerReference) bool {
+						return ownerRef.APIVersion == vectorizedv1alpha1.GroupVersion.String() && ownerRef.Kind == "Cluster"
+					}) {
+					return false, nil
+				}
+
+				instance, ok := sts.Labels["app.kubernetes.io/instance"]
```

Comment:
You can pull the cluster name from the owner reference if you change the `slices.ContainsFunc` to `slices.IndexFunc`. It'd be kinda nice to not rely on this label IMO.

Reply:
done